streamlit_extras/embed_code/__init__.py

```diff
@@ -1,267 +1,21 @@
-
-
-    Example
-    -------
-    >>> codepen_snippet("https://codepen.io/ste-vg/pen/GRooLza", width = 600, scrolling = False)
-    """
-
-    user, _, slughash = _clean_link(link).split("/")[-3:]
-    return components.html(
-        f"""
-        <p class="codepen"
-        data-height="{height}"
-        data-theme-id="{theme}"
-        data-default-tab="html,result"
-        data-user="{user}"
-        data-slug-hash="{slughash}"
-        data-preview="{str(preview).lower()}"
-        style="height: {height}px; box-sizing: border-box; display: flex; align-items: center; justify-content: center; border: 2px solid; margin: 1em 0; padding: 1em;"">
-    </p><script async src="https://static.codepen.io/assets/embed/ei.js"></script>
-    """,
-        height=height,
-        width=width,
-        scrolling=scrolling,
-    )
-
-
-def ideone_snippet(link, height=600, width=950, scrolling=True):
-    """Embed a Ideone snippet.
-
-    Parameters
-    ----------
-    link : str
-        URL from https://ideone.com/
-    height: int
-        Height of the resulting iframe
-    width: int
-        Width of the resulting iframe
-    scrolling: bool
-        If content is larger than iframe size, provide scrollbars?
-
-    Example
-    -------
-    >>> ideone_snippet("https://ideone.com/vQ54cr")
-    """
-
-    snippetnumber = _clean_link(link).split("/")[-1]
-    return components.html(
-        f"""<script src="https://ideone.com/e.js/{snippetnumber}" type="text/javascript" ></script>""",
-        height=height,
-        width=width,
-        scrolling=scrolling,
-    )
-
-
-def tagmycode_snippet(link, height=600, width=950, scrolling=True):
-    """Embed a TagMyCode snippet.
-
-    Parameters
-    ----------
-    link : str
-        URL from https://tagmycode.com/
-    height: int
-        Height of the resulting iframe
-    width: int
-        Width of the resulting iframe
-    scrolling: bool
-        If content is larger than iframe size, provide scrollbars?
-
-    Example
-    -------
-    >>> tagmycode_snippet("https://tagmycode.com/snippet/5965/recursive-list-files-in-a-dir#.Xwyc43VKglU")
-    """
-
-    snippetnumber = _clean_link(link).split("/")[-2]
-    return components.html(
-        f"""<script src="https://tagmycode.com/embed/js/{snippetnumber}"></script>""",
-        height=height,
-        width=width,
-        scrolling=scrolling,
-    )
-
-
-_SUPPORTED_PLATORMS = {
-    "github": github_gist,
-    "gitlab": gitlab_snippet,
-    "codepen": codepen_snippet,
-    "ideone": ideone_snippet,
-    "tagmycode": tagmycode_snippet,
-}
-
-
-def embed_code(
-    platform: Literal["github", "gitlab", "codepen", "ideone", "tagmycode"],
-    link: str,
-    height: int = 600,
-    width: int = 950,
-    scrolling: bool = True,
-):
-
-    validators.url(link)
-
-    return _SUPPORTED_PLATORMS[platform](
-        link=link, height=height, width=width, scrolling=scrolling
-    )
+from streamlit_embedcode import github_gist
```

Comment:
Now only supporting GitHub because extras are meant for a single function (for now!). Adding an issue to track how we can also support more functions later, so we can add back other embeds here